src/storage/src/hummock/store/version.rs

```diff
@@ -115,23 +115,27 @@ pub struct StagingVersion {
 impl StagingVersion {
     pub fn prune_overlap<'a>(
         &'a self,
-        epoch: HummockEpoch,
+        min_epoch: HummockEpoch,
+        max_epoch: HummockEpoch,
         table_id: TableId,
         key_range: &'a (Bound<Vec<u8>>, Bound<Vec<u8>>),
     ) -> (
         impl Iterator<Item = &ImmutableMemtable> + 'a,
         impl Iterator<Item = &SstableInfo> + 'a,
     ) {
         let overlapped_imms = self.imm.iter().filter(move |imm| {
-            imm.epoch() <= epoch
+            imm.epoch() <= max_epoch
+                && imm.epoch() > min_epoch
                 && range_overlap(key_range, imm.start_user_key(), imm.end_user_key())
         });
 
+        // TODO: Remove duplicate sst based on sst id
         let overlapped_ssts = self
             .sst
             .iter()
             .filter(move |staging_sst| {
-                *staging_sst.epochs.last().expect("epochs not empty") <= epoch
+                *staging_sst.epochs.last().expect("epochs not empty") <= max_epoch
+                    && *staging_sst.epochs.first().expect("epochs not empty") > min_epoch
```

Comment:
We may add a sanity check here that `max_sst_epoch <= min_epoch || min_sst_epoch > min_epoch` to ensure that `min_epoch` will not fall between `max_sst_epoch` and `min_sst_epoch`, because `min_epoch` is supposed to a committed epoch.